python/Ganga/Core/GangaRepository/Registry.py

```diff
@@ -877,9 +877,17 @@ def info(self, full=False):
                 s += ", %i other concurrent sessions:\n * %s" % (len(other_sessions), "\n * ".join(other_sessions))
         return s
 
+    def other_session_active(self):
+        """ Returns a True/False for if other sessions have been detected from the Repository """
+        if self.repository.get_other_sessions():
+            return True
+        else:
+            return False
+
     def print_other_sessions(self):
-        other_sessions = self.repository.get_other_sessions()
-        if len(other_sessions) > 0:
+        """ Prints a warning if other active sessions have been detected """
+        if self.other_session_active():
+            other_sessions = self.repository.get_other_sessions()
```

Comment:
Because it's the repository that we care about. Registries don't have 'other sessions' since they are in in-memory construct. It is the repository that can care about other processes and we want to know if there are other processes reading the files on dick. Accessing the repository from the registry is fine; it's accessing the registry from the repository that is bad in my opinion.